server/src/test/java/io/crate/types/NumericTypeTest.java

```diff
@@ -249,4 +250,11 @@ public void test_comparisons_operators_with_different_precision_and_negative_val
         assertThat(e.asSymbol("1.11::numeric(3, 1) >= 1.10::numeric(3, 2)")).isEqualTo(BOOLEAN_TRUE);
         assertThat(e.asSymbol("-2.1::numeric(5, 1) > -3.1::numeric(4, 2)")).isEqualTo(BOOLEAN_TRUE);
     }
+
+    @Test
+    public void test_cast_number_if_precision_is_lost_throws_exception() {
+        assertThatThrownBy(() -> new NumericType(6, 3).implicitCast(1234.567d))
+            .isExactlyInstanceOf(ClassCastException.class)
+            .hasMessage("Cannot cast '1234.567' to numeric(6,3) as it looses precision");
+    }
```

Comment:
Shouldn't we add an additional test case that `123.4567::numeric(6,3)` is allowed and rounds the decimal points?

Reply:
Added to the corresponding method.